### PR TITLE
Do not crash when calculating number of items in a list that has an invalid team task filter

### PR DESCRIPTION
**Pull request by caiosba**
## Description

I noticed it happening on live only for duplicated workspaces. When a list has a team task filter that doesn't have a valid ID, an exception would happen. I was able to reproduce that exception in a unit test.

The fix was to discard the team task filter if the ID is blank.

Fixes CV2-3742.

## How has this been tested?

TDD. I was able to implement a unit test that reproduced the issue.

## Checklist

- [x] I have performed a self-review of my own code
- [ ] I have added unit and feature tests, if the PR implements a new feature or otherwise would benefit from additional testing
- [x] I have added regression tests, if the PR fixes a bug
- [ ] I have added logging, exception reporting, and custom tracing with any additional information required for debugging
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [x] My changes generate no new warnings
- [ ] If I added a third party module, I included a rationale for doing so and followed our current [guidelines](https://meedan.atlassian.net/wiki/spaces/ENG/overview#Choose-the-%E2%80%9Cright%E2%80%9D-3rd-party-module)

### CODE REVIEW

File: lib/check_search.rb [hunk 1]
```diff
@@ -488,8 +488,8 @@ def file_filter
 
   def team_tasks_conditions
     conditions = []
-    return conditions unless @options.has_key?('team_tasks') && @options['team_tasks'].class.name == 'Array'
-    @options['team_tasks'].delete_if{ |tt| tt['response'].blank? }
+    return conditions unless @options['team_tasks'].class.name == 'Array'
```
Comment (melsawy) [inline lib/check_search.rb:491]: why did you remove `@options.has_key?('team_tasks')` from unless condition?
I think we should keep it and only update `delete_if` to include `tt['id'].blank?`

**Review by melsawy — CHANGES_REQUESTED** [review]

File: lib/check_search.rb (hunk 1 above)
Comment (caiosba) [reply lib/check_search.rb:491]: @melsawy because I think it's redundant with `@options['team_tasks'].class.name` ... is there is no `team_tasks` key, `@options['team_tasks'].class.name` is going to return `NilClass`

**Review by melsawy — APPROVED** [review]
